### PR TITLE
Fix: redis caching for IPFS, skip for images

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #5557 

## Description of Changes
- Handles images loaded via IPFS with a `&image` param
- Skips redis caching for images, because our cacheDecorator is not set up to store vital header info

## "How We Fixed It"

- Avatars in Snapshot descriptions will bypass redis cache (our only use case for IPFS images so far)
- Proposal metadata loaded via IPFS will be cached as before. And not returned as arraybuffer. In the bug, the cache was storing arraybuffer data instead of JSON. Since header context is not persisted to cache, the browser did not know how to handle cached responses, so it just had to deal with arraybuffers.

## Test Plan
- CA (click around) tested on local:
  - Make sure redis caching is on. You may need to clear it: `redis-cli` -> `flushall`
  - Proposals:
    - go to http://localhost:8080/dydx/proposals -> Expect titles and descriptions to lazy-load
    - inspect `ipfsProxy` network requests -> expect response header `X-Cache: Miss`
    - refresh -> expect same titles and descriptions
    - inspect `ipfsProxy` network requests -> expect response header `X-Cache: Hit`
    - repeat for http://localhost:8080/regen/proposals
  - Snapshot notification avatar
    - go to http://localhost:8080/dydx/snapshot/dydxgov.eth
    - join community and subscribe to Notifications and subscribe to Snapshots
    - In profile dropdown, click Notifications
    - Expect dydx logo to lazy-load under "Snapshot Descriptions": Screenshot 2023-11-09 at 10 01 49 PM
    - inspect `ipfsProxy` network requests -> expect response header `X-Cache: NOKEY`
    - refresh and expect the same

## Deployment Plan

1. 

## Other Considerations

- We may want to cache IPFS images in the future, but we haven't actually achieved this yet. It should be a separate effort from fixing this bug.

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/discussions/helpers.ts [hunk 1]
```diff
@@ -26,9 +27,10 @@ export const isHot = (thread: Thread) => {
 export const isNewThread = (threadCreatedAt: moment.Moment) => {
   const diffInMs = moment().diff(threadCreatedAt);
   return moment.duration(diffInMs).asHours() < 48;
-}
+};
 
 export const getLastUpdate = (thread: Thread): number => {
+  if (!thread) return 0;
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/views/pages/discussions/helpers.ts:33]: This is a fix discussed in this comment: https://github.com/hicommonwealth/commonwealth/pull/5595#discussion_r1387083555

File: packages/commonwealth/client/styles/pages/notification_settings/index.scss [hunk 2]
```diff
@@ -112,7 +112,7 @@
   }
 
   .chain-events-section-margin {
-    margin: 0 0 16px 0;
+    margin: 30px 0 16px 0;
```
Comment (mhagel) [inline packages/commonwealth/client/styles/pages/notification_settings/index.scss:115]: I noticed this section's margin did not match the others on the page, so fixing it here

**Review by lzach83 — APPROVED** [review]
Tested locally, works as intended. Great job.

**Comment by mhagel** [comment]
>@mhagel I don't quite understand why we need to not cache images 


As mentioned:
> We may want to cache IPFS images in the future, but we haven't actually achieved this yet. It should be a separate effort from fixing this bug.

> -- can't we cache the content-type as well to ensure we return the correct object?

The cache middleware is set up to intercept and cache the response data, and not the headers. I didn't want to rewrite it just to handle the image case. Currently, we aren't caching images anyway - it's broken too. But luckily there is only one small case where we try so far: avatars in snapshot notifications.

@jnaviask

**Comment by mhagel** [comment]
Follow-up: https://github.com/hicommonwealth/commonwealth/issues/5806

**Review by kurtisassad — APPROVED** [review]
LGTM

**Comment by mhagel** [comment]
FWIW, this also fixes a bug where  image load fails on refresh (because the cache is not working as intended). We will make it load faster in the follow-up noted above.

**Pull request merged**